diff --git a/registry.js b/registry.js
new file mode 100644
index 0000000000000000000000000000000000000000..8bf537f37f2d0b877dcf57bf38e17dda3edb47b0
--- /dev/null
+++ b/registry.js
@@ -0,0 +1,25 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow strict
+ * @format
+ */
+
+"use strict";
+
+const assets = [];
+
+function registerAsset(asset) {
+  // `push` returns new array length, so the first asset will
+  // get id 1 (not 0) to make the value truthy
+  return assets.push(asset);
+}
+
+function getAssetByID(assetId) {
+  return assets[assetId - 1];
+}
+
+module.exports = { registerAsset, getAssetByID };
