diff --git a/node_modules/@juspay-tech/react-native-hyperswitch-netcetera-3ds/react-native-hyperswitch-netcetera-3ds.podspec b/node_modules/@juspay-tech/react-native-hyperswitch-netcetera-3ds/react-native-hyperswitch-netcetera-3ds.podspec
index 1c89356..9102d93 100644
--- a/node_modules/@juspay-tech/react-native-hyperswitch-netcetera-3ds/react-native-hyperswitch-netcetera-3ds.podspec
+++ b/node_modules/@juspay-tech/react-native-hyperswitch-netcetera-3ds/react-native-hyperswitch-netcetera-3ds.podspec
@@ -28,7 +28,6 @@ Pod::Spec.new do |s|
     }
 
     s.dependency "React-Codegen"
-    s.dependency "RCT-Folly"
     s.dependency "RCTRequired"
     s.dependency "RCTTypeSafety"
     s.dependency "ReactCommon/turbomodule/core"
