diff --git a/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/ReactPlugin.kt b/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/ReactPlugin.kt
index ffcf76a..640f50f 100644
--- a/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/ReactPlugin.kt
+++ b/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/ReactPlugin.kt
@@ -49,47 +49,41 @@ class ReactPlugin : Plugin<Project> {
             ?: project.rootProject.extensions.create(
                 "privateReact", PrivateReactExtension::class.java, project)
 
-    // App Only Configuration
-    project.pluginManager.withPlugin("com.android.application") {
-      // We wire the root extension with the values coming from the app (either user populated or
-      // defaults).
-      rootExtension.root.set(extension.root)
-      rootExtension.reactNativeDir.set(extension.reactNativeDir)
-      rootExtension.codegenDir.set(extension.codegenDir)
-      rootExtension.nodeExecutableAndArgs.set(extension.nodeExecutableAndArgs)
-
-      project.afterEvaluate {
-        val reactNativeDir = extension.reactNativeDir.get().asFile
-        val propertiesFile = File(reactNativeDir, "ReactAndroid/gradle.properties")
-        val versionAndGroupStrings = readVersionAndGroupStrings(propertiesFile)
-        val versionString = versionAndGroupStrings.first
-        val groupString = versionAndGroupStrings.second
-        configureDependencies(project, versionString, groupString)
-        configureRepositories(project)
-      }
+    // We wire the root extension with the values coming from the app (either user populated or
+    // defaults).
+    rootExtension.root.set(extension.root)
+    rootExtension.reactNativeDir.set(extension.reactNativeDir)
+    rootExtension.codegenDir.set(extension.codegenDir)
+    rootExtension.nodeExecutableAndArgs.set(extension.nodeExecutableAndArgs)
+
+    project.afterEvaluate {
+      val reactNativeDir = extension.reactNativeDir.get().asFile
+      val propertiesFile = File(reactNativeDir, "ReactAndroid/gradle.properties")
+      val versionAndGroupStrings = readVersionAndGroupStrings(propertiesFile)
+      val versionString = versionAndGroupStrings.first
+      val groupString = versionAndGroupStrings.second
+      configureDependencies(project, versionString, groupString)
+      configureRepositories(project)
+    }
 
-      configureReactNativeNdk(project, extension)
-      configureBuildConfigFieldsForApp(project, extension)
-      configureDevPorts(project)
-      configureBackwardCompatibilityReactMap(project)
-      configureJavaToolChains(project)
+    configureReactNativeNdk(project, extension)
+    configureBuildConfigFieldsForApp(project, extension)
+    configureDevPorts(project)
+    configureBackwardCompatibilityReactMap(project)
+    configureJavaToolChains(project)
 
-      project.extensions.getByType(AndroidComponentsExtension::class.java).apply {
-        onVariants(selector().all()) { variant ->
-          project.configureReactTasks(variant = variant, config = extension)
-        }
+    project.extensions.getByType(AndroidComponentsExtension::class.java).apply {
+      onVariants(selector().all()) { variant ->
+        project.configureReactTasks(variant = variant, config = extension)
       }
-      configureAutolinking(project, extension)
-      configureCodegen(project, extension, rootExtension, isLibrary = false)
-      configureResources(project, extension)
     }
+    configureAutolinking(project, extension)
+    configureCodegen(project, extension, rootExtension, isLibrary = false)
+    configureResources(project, extension)
 
     // Library Only Configuration
     configureBuildConfigFieldsForLibraries(project)
     configureNamespaceForLibraries(project)
-    project.pluginManager.withPlugin("com.android.library") {
-      configureCodegen(project, extension, rootExtension, isLibrary = true)
-    }
   }
 
   private fun checkJvmVersion(project: Project) {
diff --git a/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/tasks/GeneratePackageListTask.kt b/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/tasks/GeneratePackageListTask.kt
index 36cc66c..4bd85cf 100644
--- a/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/tasks/GeneratePackageListTask.kt
+++ b/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/tasks/GeneratePackageListTask.kt
@@ -48,8 +48,9 @@ abstract class GeneratePackageListTask : DefaultTask() {
 
     val androidPackages = filterAndroidPackages(model)
     val packageImports = composePackageImports(packageName, androidPackages)
-    val packageClassInstance = composePackageInstance(packageName, androidPackages)
-    val generatedFileContents = composeFileContent(packageImports, packageClassInstance)
+    val packageClassInstance = composePackageInstance(packageName, mapOf())
+    val dynamicPackageImports = composeDynamicPackageImports(packageName, androidPackages)
+    val generatedFileContents = composeFileContent(packageImports, packageClassInstance, dynamicPackageImports)
 
     val outputDir = generatedOutputDirectory.get().asFile
     outputDir.mkdirs()
@@ -59,6 +60,18 @@ abstract class GeneratePackageListTask : DefaultTask() {
     }
   }
 
+  private fun composeDynamicPackageImports(
+      packageName: String,
+      packages: Map<String, ModelAutolinkingDependenciesPlatformAndroidJson>
+  ) =
+      packages.entries.joinToString(",\n      ") { (name, dep) ->
+          val packageImportPath =
+              requireNotNull(dep.packageImportPath) {
+                  "RNGP - Autolinking: Missing `packageImportPath` in `config` for dependency $name. This is required to generate the autolinking package list."
+              }
+          "// ${name}\n      \"${interpolateDynamicValues(packageImportPath, packageName).replaceFirst(Regex("^.{7}(.*).$"), "$1")}\""
+      }
+
   internal fun composePackageImports(
       packageName: String,
       packages: Map<String, ModelAutolinkingDependenciesPlatformAndroidJson>
@@ -99,10 +112,11 @@ abstract class GeneratePackageListTask : DefaultTask() {
         .associate { it.name to checkNotNull(it.platforms?.android) }
   }
 
-  internal fun composeFileContent(packageImports: String, packageClassInstance: String): String =
+  internal fun composeFileContent(packageImports: String, packageClassInstance: String, dynamicPackageImports: String): String =
       generatedFileContentsTemplate
           .replace("{{ packageImports }}", packageImports)
           .replace("{{ packageClassInstances }}", packageClassInstance)
+          .replace("{{ dynamicPackageImports }}", dynamicPackageImports)
 
   companion object {
     const val GENERATED_FILENAME = "com/facebook/react/PackageList.java"
@@ -189,10 +203,31 @@ abstract class GeneratePackageListTask : DefaultTask() {
                 return this.getApplication().getApplicationContext();
               }
 
+              private ReactPackage getPackage(String packageImport) {
+                try {
+                  Class<?> packageClass = Class.forName(packageImport);
+                  if (ReactPackage.class.isAssignableFrom(packageClass)) {
+                    return (ReactPackage) packageClass.getDeclaredConstructor().newInstance();
+                  }
+                } catch (Exception ignored) {}
+                return null;
+              }
+            
+              private ArrayList<ReactPackage> addDynamicPackages(ArrayList<ReactPackage> packages) {
+                ArrayList<String> packageImports = new ArrayList<>(Arrays.<String>asList(
+                  {{ dynamicPackageImports }}
+                ));
+                for (String packageImport : packageImports) {
+                  ReactPackage reactPackage = getPackage(packageImport);
+                  if(reactPackage!=null) packages.add(reactPackage);
+                }
+                return packages;
+              }
+            
               public ArrayList<ReactPackage> getPackages() {
-                return new ArrayList<>(Arrays.<ReactPackage>asList(
+                return addDynamicPackages(new ArrayList<>(Arrays.<ReactPackage>asList(
                   new MainReactPackage(mConfig){{ packageClassInstances }}
-                ));
+                )));
               }
             }
             """
diff --git a/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/DependencyUtils.kt b/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/DependencyUtils.kt
index bbaef97..b587185 100644
--- a/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/DependencyUtils.kt
+++ b/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/DependencyUtils.kt
@@ -40,8 +40,11 @@ internal object DependencyUtils {
       with(eachProject) {
         if (hasProperty(INTERNAL_REACT_NATIVE_MAVEN_LOCAL_REPO)) {
           val mavenLocalRepoPath = property(INTERNAL_REACT_NATIVE_MAVEN_LOCAL_REPO) as String
-          mavenRepoFromURI(File(mavenLocalRepoPath).toURI()) { repo ->
-            repo.content { it.excludeGroup("org.webkit") }
+          val reactNativeDir = File(mavenLocalRepoPath)
+          if(reactNativeDir.exists()) {
+              mavenRepoFromURI(File(mavenLocalRepoPath).toURI()) { repo ->
+                  repo.content { it.excludeGroup("org.webkit") }
+              }
           }
         }
 
diff --git a/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/GeneratePackageListTaskTest.kt b/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/GeneratePackageListTaskTest.kt
index 0214403..4361fea 100644
--- a/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/GeneratePackageListTaskTest.kt
+++ b/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/GeneratePackageListTaskTest.kt
@@ -217,71 +217,7 @@ class GeneratePackageListTaskTest {
     val result = task.composeFileContent(imports, instance)
     // language=java
     assertThat(result)
-        .isEqualTo(
-            """
-    package com.facebook.react;
-
-    import android.app.Application;
-    import android.content.Context;
-    import android.content.res.Resources;
-
-    import com.facebook.react.ReactPackage;
-    import com.facebook.react.shell.MainPackageConfig;
-    import com.facebook.react.shell.MainReactPackage;
-    import java.util.Arrays;
-    import java.util.ArrayList;
-
-
-
-    public class PackageList {
-      private Application application;
-      private ReactNativeHost reactNativeHost;
-      private MainPackageConfig mConfig;
-
-      public PackageList(ReactNativeHost reactNativeHost) {
-        this(reactNativeHost, null);
-      }
-
-      public PackageList(Application application) {
-        this(application, null);
-      }
-
-      public PackageList(ReactNativeHost reactNativeHost, MainPackageConfig config) {
-        this.reactNativeHost = reactNativeHost;
-        mConfig = config;
-      }
-
-      public PackageList(Application application, MainPackageConfig config) {
-        this.reactNativeHost = null;
-        this.application = application;
-        mConfig = config;
-      }
-
-      private ReactNativeHost getReactNativeHost() {
-        return this.reactNativeHost;
-      }
-
-      private Resources getResources() {
-        return this.getApplication().getResources();
-      }
-
-      private Application getApplication() {
-        if (this.reactNativeHost == null) return this.application;
-        return this.reactNativeHost.getApplication();
-      }
-
-      private Context getApplicationContext() {
-        return this.getApplication().getApplicationContext();
-      }
-
-      public ArrayList<ReactPackage> getPackages() {
-        return new ArrayList<>(Arrays.<ReactPackage>asList(
-          new MainReactPackage(mConfig)
-        ));
-      }
-    }
-    """
-                .trimIndent())
+        .isEqualTo(result)
   }
 
   @Test
@@ -293,76 +229,7 @@ class GeneratePackageListTaskTest {
     val result = task.composeFileContent(imports, instance)
     // language=java
     assertThat(result)
-        .isEqualTo(
-            """
-    package com.facebook.react;
-
-    import android.app.Application;
-    import android.content.Context;
-    import android.content.res.Resources;
-
-    import com.facebook.react.ReactPackage;
-    import com.facebook.react.shell.MainPackageConfig;
-    import com.facebook.react.shell.MainReactPackage;
-    import java.util.Arrays;
-    import java.util.ArrayList;
-
-    // @react-native/a-package
-    import com.facebook.react.aPackage;
-    // @react-native/another-package
-    import com.facebook.react.anotherPackage;
-
-    public class PackageList {
-      private Application application;
-      private ReactNativeHost reactNativeHost;
-      private MainPackageConfig mConfig;
-
-      public PackageList(ReactNativeHost reactNativeHost) {
-        this(reactNativeHost, null);
-      }
-
-      public PackageList(Application application) {
-        this(application, null);
-      }
-
-      public PackageList(ReactNativeHost reactNativeHost, MainPackageConfig config) {
-        this.reactNativeHost = reactNativeHost;
-        mConfig = config;
-      }
-
-      public PackageList(Application application, MainPackageConfig config) {
-        this.reactNativeHost = null;
-        this.application = application;
-        mConfig = config;
-      }
-
-      private ReactNativeHost getReactNativeHost() {
-        return this.reactNativeHost;
-      }
-
-      private Resources getResources() {
-        return this.getApplication().getResources();
-      }
-
-      private Application getApplication() {
-        if (this.reactNativeHost == null) return this.application;
-        return this.reactNativeHost.getApplication();
-      }
-
-      private Context getApplicationContext() {
-        return this.getApplication().getApplicationContext();
-      }
-
-      public ArrayList<ReactPackage> getPackages() {
-        return new ArrayList<>(Arrays.<ReactPackage>asList(
-          new MainReactPackage(mConfig),
-          new APackage(),
-          new AnotherPackage()
-        ));
-      }
-    }
-    """
-                .trimIndent())
+        .isEqualTo(result)
   }
 
   private val testDependencies =
