diff --git a/node_modules/@react-native/gradle-plugin/build.gradle.kts b/node_modules/@react-native/gradle-plugin/build.gradle.kts
index aba3649..1fe4493 100644
--- a/node_modules/@react-native/gradle-plugin/build.gradle.kts
+++ b/node_modules/@react-native/gradle-plugin/build.gradle.kts
@@ -13,6 +13,7 @@ import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 plugins {
   kotlin("jvm") version "1.7.22"
   id("java-gradle-plugin")
+  id("maven-publish")
 }
 
 repositories {
@@ -30,6 +31,7 @@ gradlePlugin {
 }
 
 group = "com.facebook.react"
+version = "0.72.11"
 
 dependencies {
   implementation(gradleApi())
@@ -70,3 +72,12 @@ tasks.withType<Test>().configureEach {
     showStackTraces = true
   }
 }
+
+publishing {
+  repositories {
+    maven {
+      name = "react-native-gradle-plugin"
+      url = uri("${project.rootDir}/maven")
+    }
+  }
+}
\ No newline at end of file
diff --git a/node_modules/@react-native/gradle-plugin/src/main/kotlin/com/facebook/react/ReactPlugin.kt b/node_modules/@react-native/gradle-plugin/src/main/kotlin/com/facebook/react/ReactPlugin.kt
index 062dcc7..69a748d 100644
--- a/node_modules/@react-native/gradle-plugin/src/main/kotlin/com/facebook/react/ReactPlugin.kt
+++ b/node_modules/@react-native/gradle-plugin/src/main/kotlin/com/facebook/react/ReactPlugin.kt
@@ -77,6 +77,33 @@ class ReactPlugin : Plugin<Project> {
 
     // Library Only Configuration
     project.pluginManager.withPlugin("com.android.library") {
+      // We wire the root extension with the values coming from the app (either user populated or
+      // defaults).
+      rootExtension.root.set(extension.root)
+      rootExtension.reactNativeDir.set(extension.reactNativeDir)
+      rootExtension.codegenDir.set(extension.codegenDir)
+      rootExtension.nodeExecutableAndArgs.set(extension.nodeExecutableAndArgs)
+
+      project.afterEvaluate {
+        val reactNativeDir = extension.reactNativeDir.get().asFile
+        val propertiesFile = File(reactNativeDir, "ReactAndroid/gradle.properties")
+        val versionAndGroupStrings = readVersionAndGroupStrings(propertiesFile)
+        val versionString = versionAndGroupStrings.first
+        val groupString = versionAndGroupStrings.second
+        configureDependencies(project, versionString, groupString)
+        configureRepositories(project, reactNativeDir)
+      }
+
+      configureReactNativeNdk(project, extension)
+      configureBuildConfigFields(project)
+      configureDevPorts(project)
+      configureBackwardCompatibilityReactMap(project)
+
+      project.extensions.getByType(AndroidComponentsExtension::class.java).apply {
+        onVariants(selector().all()) { variant ->
+          project.configureReactTasks(variant = variant, config = extension)
+        }
+      }
       configureCodegen(project, extension, rootExtension, isLibrary = true)
     }
 
diff --git a/node_modules/@react-native/gradle-plugin/src/main/kotlin/com/facebook/react/utils/DependencyUtils.kt b/node_modules/@react-native/gradle-plugin/src/main/kotlin/com/facebook/react/utils/DependencyUtils.kt
index ecf0830..f0acd5a 100644
--- a/node_modules/@react-native/gradle-plugin/src/main/kotlin/com/facebook/react/utils/DependencyUtils.kt
+++ b/node_modules/@react-native/gradle-plugin/src/main/kotlin/com/facebook/react/utils/DependencyUtils.kt
@@ -35,7 +35,9 @@ internal object DependencyUtils {
           repo.content { it.excludeModule("org.webkit", "android-jsc") }
         }
         // Android JSC is installed from npm
-        mavenRepoFromURI(File(reactNativeDir, "../jsc-android/dist").toURI())
+        if (reactNativeDir.exists()) {
+          mavenRepoFromURI(File(reactNativeDir, "../jsc-android/dist").toURI())
+        }
         repositories.google()
         mavenRepoFromUrl("https://www.jitpack.io")
       }
