diff --git a/node_modules/react-native-code-push/android/codepush.gradle b/node_modules/react-native-code-push/android/codepush.gradle
index 4faf49f..2613a5f 100644
--- a/node_modules/react-native-code-push/android/codepush.gradle
+++ b/node_modules/react-native-code-push/android/codepush.gradle
@@ -26,7 +26,10 @@ android.buildTypes.each { buildType ->
 gradle.projectsEvaluated {
     def debuggableVariants = config.debuggableVariants.get() ?: ['debug']
 
-    android.applicationVariants.all { variant ->
+    def variants = android.hasProperty('applicationVariants') ?
+            android.applicationVariants : android.libraryVariants
+    
+    variants.all { variant ->
         // No code push for debuggable variants
         if (debuggableVariants.contains(variant.name)) {
             return;
